addon/components/link-to-component.js

```diff
@@ -15,6 +16,11 @@ export default LinkComponent.extend({
 
     let owner = getOwner(this);
 
+    assert(
+      `You attempted to use a link-to within a routeless engine, this is not supported.`,
```

Comment:
Does the `owner` have easy access to it? If it's not straightforward, we can worry about it if it becomes an issue.